### ISSUE TITLE
Drawio keeps crashing

### ISSUE THREAD

**Issue by markofemu** [jgraph/drawio-desktop#1758]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio-desktop/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

You must agree to search and the code of conduct. You must fill in this entire template. If you delete part/all or miss parts out your issue will be closed.

**Describe the bug**

I have a 20 MB Drawio file which expands to 2.6 GB when loaded in.  It is a (very large) family tree I am compiling and uses the vertical tree structure of parent branching to children, grandchildren, etc.  (I am actually using it as child branching to parents, grandparents, etc.).  
Every time I add a couple of boxes (for another set of parents), many of the boxes automatically move their positions slightly, and after about ten boxes have been added, RAM usage (as given by Windows Task Manager) has increased to around 6 or 7 GB.  This is presumably because “Undo” has to keep a record of all the thousands of boxes newest positions every time a new box is added. 
If I keep adding and editing boxes Drawio will crash, usually once it is using anywhere between 6 and 8 GB.  I get a white Drawio screen, which I can’t close, but have to use Windows Task Manager to end the Drawio task.  
When I go to open the file again, Drawio opens to the 2.6 GB again, but if it has crashed without edits having been saved, Drawio also loads a backup and asks me whether I want to use the unsaved changes (I always discard these, as I don’t know what might have happened to my file during the crash).  Unfortunately, the backup is also loaded into RAM, thereby doubling the RAM in use by Drawio.  Even though I discard the backup changes, the memory it is using is not released, so my new editing starts with around 5 GB of RAM in use.  Just adding a few boxes or doing a search/replace will take me to 6 or 7 GB RAM usage, and another crash will occur.  
One way to stop the continual crashing is to have Windows Task Manager running all the time and keep a check on RAM usage, and once it exceeds 6 GB (about every 5 minutes of editing), save the edit, exit Drawio and start again.  This is a very inefficient way of editing my file.  

**To Reproduce**

(I need to upload the Drawio file, where the addition of a dozen new boxes will reproduce the problem, but if I try to add it you say "We don’t support that file type".)

**Expected behavior**

Answers to:
Can Drawio be altered to use all available RAM on the computer?
If not, what is the limit on expanded file size for Drawio?  
Although I have 40 GB of RAM, is there some reason I can’t use more than 6 to 8 GB of RAM for Drawio?  
Can I stop it loading the backup, or if not can I release the memory the discarded backup is using?  

**Screenshots**

**draw.io version (In the Help->About menu of the draw.io editor):**

- draw.io version 24.4.8 

**Desktop (please complete the following information):**

 - OS: Windows 11 Pro, version 23H2

**Additional context**

**Comment by davidjgraph** [comment]
zip the file and attach it.

**Comment by markofemu** [comment]
[Family tree1.zip](https://github.com/jgraph/drawio-desktop/files/15468076/Family.tree1.zip)

Thank you David.  Please find attached the file in question (I have removed some of the most recent generations at the top for privacy / security)

**Comment by davidjgraph** [comment]
OK, that probably is the largest draw.io file I've seen in 14 years, I feel you deserve some kind of award for creating a 220k line file.

We'll take a look.

**Comment by davidjgraph** [comment]
You could disable extras->automatic backup and set the draft interval to 0 to disable, but would then need to very cafeful to save often.

**Comment by davidjgraph** [comment]
And untick extras->autosave

**Comment by markofemu** [comment]
I already have autosave unticked, and have previously tried disabling automatic backup in the past, but without success.  I haven't set the draft interval to 0 though.  So I'll try that too.  BTW, I have learnt to manually save after adding each pair of boxes (for another generation of parents).  
Incidentally, adding boxes uses up far more RAM that editing the contents of a box, presumably because all the newly shuffled box positions use a lot of memory, compared to a few keystrokes for a name change.

**Comment by markofemu** [comment]
Hi David,
I did all three (unticked autosave, disabled automatic backup and set draft interval to 0).  It still crashed around the 6 to 8 GB mark, but when I killed the task in Windows Task Manager and restarted Drawio, thankfully it didn't try and load a backup.

**Comment by davidjgraph** [comment]
What about spliting sections into multiple pages and linking nodes into other pages? The problem here is the sheer volume of revalidation and drawing when one thing changes.

**Comment by markofemu** [comment]
I did consider splitting into multiple files, but I have to keep searching on names to check for repeats of parts of the family tree, where two or more children appear, but have the same parents.  If the parents' family trees are already done, I don't want to do them again (especially since editing is so slow - to create a new box takes around 20 seconds, and a similar time to change the name in it).  
I now see that a search will do so over multiple pages of the same file.  
Ultimately it will be printed on one long sheet, so can the multiple pages be rejoined into one page when I have finished?
